inference-engine/src/vpu/graph_transformer/src/middleend/special_stage_processor.cpp

```diff
@@ -5,99 +5,249 @@
+                    IE_ASSERT(input->checkStrides(consumerStrideReqs));
 
-                        if (!checkStrides(output->desc(), input->strides(), consumerStrideReqs)) {
-                            needCopy = true;
-                            break;
-                        }
+                    if (!checkStrides(input->desc(), outputData->strides(), consumerStrideReqs)) {
+                        isCopyNeed = true;
+                        isCopyOptional = false;
                     }
                 }
             }
         }
 
         //
-        // Insert Copy if needed
+        // Check producer StridesRequirement.
         //
 
-        if (needCopy) {
-            auto outputCopy = model->duplicateData(output, "@copy");
-            outputCopy->resetRequiredStrides();
+        if (!isCopyNeed) {
+            if (auto producerEdge = input->producerEdge()) {
+                const auto& producerInfo = producerEdge->producer()->getDataStridesRequirements();
 
-            auto outPortInd = outEdge->portInd();
+                if (producerInfo.hasOutput(producerEdge)) {
+                    const auto& producerStrideReqs = producerInfo.getOutput(producerEdge);
+                    IE_ASSERT(input->checkStrides(producerStrideReqs));
 
-            model->replaceStageOutput(outEdge, outputCopy);
+                    if (!checkStrides(input->desc(), outputData->strides(), producerStrideReqs)) {
+                        isCopyNeed = true;
+                        isCopyOptional = false;
+                    }
+                }
 
-            auto copyStage = _stageBuilder->addCopyStage(
-                model,
-                formatString("%s@output=%d@copy-for-split", stage->name(), outPortInd),
-                stage->origLayer(),
-                outputCopy,
-                output,
-                "special::split");
-            if (stage->attrs().has("batchInd")) {
-                copyStage->attrs().set("batchInd", stage->attrs().get<int>("batchInd"));
+                if (!isCopyNeed) {
+                    //
+                    // To reduce the size of HW output (still can be optimized).
+                    //
+
+                    if (producerEdge->producer()->category() == StageCategory::HW) {
+                        isCopyNeed = true;
+                        isCopyOptional = true;
+                    }
+                }
             }
+        }
+    }
 
-            output = outputCopy;
+    NeedCopyDesc needCopyDesc;
+    needCopyDesc.isCopyNeed = isCopyNeed;
+    if (stage->type() == StageType::StubConcat ||
+        stage->type() == StageType::Expand) {
+        needCopyDesc.isCopyOptional = isCopyOptional;
+    }
+    return needCopyDesc;
+}
+
+Data insertCopyOfInput(
+                const Model& model,
+                const Stage& stage,
+                const StageInput& edge,
+                const StageBuilder::Ptr& _stageBuilder,
+                const char* stageType,
```

Comment:
You don't need to pass this parameter, you can extract information from `stage`

Reply:
I don't know any way to get stage type string from `stage`. Could you help me with it, please?

Reply:
First of all, there is a chance you don't need to have `std::string` representation of `StageType` explicitly, since it has output stream operator overloaded. If it does not help you always can use this operator to get `std::string` representation:

```cpp
std::stringstream typeAsString;
typeAsString << stage->type();
// use typeAsString.str()
```

Reply:
Fixed